### PR TITLE
Authentication with video api call

### PR DESCRIPTION
**Pull request by animeshgarai09**
Hey there folks!
Added authentication flow with API calls.
Started fetching video details from the backend.

You can review the store folder where context and reducers are kept. 
Thanks😃

### CODE REVIEW

File: src/components/header/header.module.scss [hunk 1]
```diff
@@ -44,6 +53,9 @@
 .dropdown {
     top: 4.2rem !important;
```
Comment (MayankKumar10) [inline src/components/header/header.module.scss:54]: try to avoid use of important tag

**Review by MayankKumar10 — APPROVED** [review]
Just a suggestion everything else looks good to me, UI Looks good, you can merge it.

File: src/store/reducers/video-reducer.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { videoCountByCategory } from "./reducer-functions"
+export const VideoReducer = (state, action) => {
+    switch (action.type) {
+        case "SET_VIDEOS_&_CATEGORIES":
+            return { ...state, videos: action.payload.videos, categories: videoCountByCategory(action.payload.videos, action.payload.categories) }
```
Comment (ShekharDhangar) [inline src/store/reducers/video-reducer.js:5]: you can destructure in parameters like (state,{type , payload})

**Review by aayushjain-code — APPROVED** [review]
Very nice UI use of scss makes things much better go ahead and merge it

File: src/store/providers/video-provider.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        Promise.all(endpoints.map((endpoint) => axios.get(endpoint))).then(
+            axios.spread((...allData) => {
+                const [videos, categories] = allData
+
+                VideoDispatch({
+                    type: "SET_VIDEOS_&_CATEGORIES",
+                    payload: {
+                        videos: videos.data.videos,
+                        categories: categories.data.categories
+                    }
```
Comment (ShekharDhangar) [inline src/store/providers/video-provider.js:27]: Liked how you do axios.spread. Will try that

**Review by ShekharDhangar — APPROVED** [review]
Rest looks good,try to look into suggested change.

File: src/store/providers/video-provider.js (hunk 3 above)
Comment (ShekharDhangar) [reply src/store/providers/video-provider.js:27]: Also one question why not use async await?

**Pull request merged**